File: javav2/example_code/glue/src/main/java/com/example/glue/SearchTables.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.glue;

// snippet-start:[glue.java2.search_table.main]
// snippet-start:[glue.java2.search_table.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.glue.GlueClient;
import software.amazon.awssdk.services.glue.model.GlueException;
import software.amazon.awssdk.services.glue.model.SearchTablesRequest;
import software.amazon.awssdk.services.glue.model.SearchTablesResponse;
import software.amazon.awssdk.services.glue.model.Table;
import java.util.List;
// snippet-end:[glue.java2.search_table.import]

/*
*   Before running this example, run a crawler to produce a table within a database.
*
*  Also, set up your development environment, including your credentials.
*
*  For more information, see the following documentation topic:
*
*  https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
*/

public class SearchTables {
    public static void main(String[] args) {
        final String usage = """

                Usage:
                    <text>

                Where:
                    text - A string used for a text search.\s
                """;

        if (args.length != 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String text = args[0];
        Region region = Region.US_EAST_1;
        GlueClient glueClient = GlueClient.builder()
                .region(region)
                .build();

        searchGlueTable(glueClient, text);
        glueClient.close();
    }

    public static void searchGlueTable(GlueClient glueClient, String text) {
        try {
            SearchTablesRequest tablesRequest = SearchTablesRequest.builder()
                    .searchText(text)
                    .resourceShareType("ALL")
                    .maxResults(10)
                    .build();

            SearchTablesResponse tablesResponse = glueClient.searchTables(tablesRequest);
            List<Table> tables = tablesResponse.tableList();
            for (Table table : tables) {
                System.out.println("Table name is : " + table.name());
                System.out.println("Database name is : " + table.databaseName());
            }

        } catch (GlueException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
    }
}
// snippet-end:[glue.java2.search_table.main]

File: kotlin/services/glue/src/main/kotlin/com/kotlin/glue/SearchTables.kt
// snippet-sourcedescription:[SearchTables.kt demonstrates how to search a set of tables based on properties.]
// snippet-keyword:[AWS SDK for Kotlin]
// snippet-keyword:[AWS Glue]
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.glue

// snippet-start:[glue.kotlin.search_table.import]
import aws.sdk.kotlin.services.glue.GlueClient
import aws.sdk.kotlin.services.glue.model.ResourceShareType
import aws.sdk.kotlin.services.glue.model.SearchTablesRequest
import kotlin.system.exitProcess
// snippet-end:[glue.kotlin.search_table.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <text>
    Where:
        text - a string used for a text search. 
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val text = args[0]
    searchGlueTable(text)
}

// snippet-start:[glue.kotlin.search_table.main]
suspend fun searchGlueTable(text: String?) {

    val request = SearchTablesRequest {
        searchText = text
        resourceShareType = ResourceShareType.fromValue("All")
        maxResults = 10
    }

    GlueClient { region = "us-west-2" }.use { glueClient ->
        val response = glueClient.searchTables(request)
        response.tableList?.forEach { table ->
            println("Table name is ${table.name}")
            println("Database name is ${table.databaseName}")
        }
    }
}
// snippet-end:[glue.kotlin.search_table.main]
